### DISCUSSION TITLE
OpenAPI Initiative 'Arazzo' : shoud CAMARA adopt it?

### DISCUSSION THREAD

**Discussion by Kevsy** [camaraproject/Commonalities#309] [General]
On today's TSC I presented on OpenAPI Initiative's [Arazzo](https://www.openapis.org/arazzo). This is a new specification, similar to OAS, that defines tests and workflows for APIs - and supports API calls across multiple distinct APIs. 

Please take a look at the attached presentation:
[TSC 2024-10-03 Intro to OAS Arazzo .pdf](https://github.com/user-attachments/files/17245680/TSC.2024-10-03.Intro.to.OAS.Arazzo.pdf) 

...and discuss below whether:
1) You see value to CAMARA to adopt Arazzo
2) What would be the reason(s):
 a) helping to write user stories through workflows
 b) reusing tests across APIs
 c) defining complex tests that span multiple CAMARA APIs (e.g. in an API family)
 d) defining error tests
3) Any advantage/disadvantage compared to Gherkin

In case of further questions, I can ask the Arazzo leads.

**Comment by Kevsy** [comment]
For point 3, I can see advantages to using Arazzo vs Gherkin:
- it has familiar syntax (based on OAS)
- the workflows feature can enable definition and testing of user stories that involve calls to several endpoints (e.g. those in Quality on Demand)
- overall Arazzo seems a closer fit to OAS APIs as it is produced by the same team

**Reply by bigludo7** [reply]
Probably good for this specific point to collect @trehman-gsma feedback. As Toyeeb and GSMA are using CAMARA test definition to produce certification process they are 'customer' of these test definition.
